### PR TITLE
Mood - sessions creation and get_all endpoints complete

### PR DESCRIPTION
**Pull request by Chulstro**
# SOM-Timer BE Pull Request Form

## Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
`Added endpoints to complete mood functionality`

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Has This Been Tested?
- [ ] No
- [x] Yes

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:
`Two tests to run checks on api response values`

# Checklist:
- [x] My code has no unused/commented out code
- [x] My code has no console logs/pry sessions
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -2,13 +2,31 @@
...
 from flask_sqlalchemy import SQLAlchemy
-from sqlalchemy import MetaData
+from sqlalchemy import MetaData, create_engine
 import os
+from sqlalchemy.orm import sessionmaker
 
 app = create_app("development")
 db = SQLAlchemy(app)
 
+@app.cli.command('db_reset')
```
Comment (sciencefixion) [inline app.py:15]: I was trying to figure something like this out for a while! Excellent job!

**Review by sciencefixion — APPROVED** [review]

File: application/controllers/rests.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+            rest = rest.all()
+
+            return marshal({
+                'count': len(rest),
+                'rests': [marshal(t, rest_fields) for t in rest]
+            }, rest_list_fields)
+
+    @marshal_with(rest_fields)
+    def post(self):
```
Comment (rwilliams659) [inline application/controllers/rests.py:85]: Thanks for spinning this up so quick! Would you mind updating the documentation for us as well when you have a second? Given my unfamiliarity with those code, it's a little hard for me to tell if the endpoint is just the root /rests or something else.

File: tests/test_rests.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+import unittest
```
Comment (rwilliams659) [inline tests/test_rests.py:1]: Yay, you got some tests up!

**Review by rwilliams659 — APPROVED** [review]
Thanks so much for whipping this up! We're about ready to hook it up!

File: app.py (hunk 1 above)
Comment (sienna-kopf) [reply app.py:15]: I LOVE this db_reset method!!! Super smart @Chulstro

File: application/__init__.py [hunk 4]
```diff
@@ -34,11 +34,13 @@ def handle_error(e):
     api.prefix = '/api'
 
     from application.controllers.timers import TimersResource
+    from application.controllers.rests import RestsResource
     from application.controllers.exercises import ExercisesResource, RandExercises
 
     api.add_resource(TimersResource, '/timers', '/timers/<int:timer_id>')
     api.add_resource(ExercisesResource, '/exercises', '/exercises/<int:exercise_id>')
     api.add_resource(RandExercises, '/rand_exercise')
+    api.add_resource(RestsResource, '/rests')
```
Comment (sienna-kopf) [inline application/__init__.py:43]: Why is this called rests? I have no problem with it, just wondering.

File: application/controllers/rests.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
+from flask_restful import Resource, reqparse, request
+from flask_restful import fields, marshal_with, marshal
+from application.models.rest import Rest
+from application import db
+
+rest_fields = {
```
Comment (sienna-kopf) [inline application/controllers/rests.py:6]: It makes a lot of sense to me to consolidate all of these attributes into one object. Nice work figuring out this data flow with the FE team @Chulstro

File: application/controllers/rests.py (hunk 2 above)
Comment (sienna-kopf) [reply application/controllers/rests.py:85]: I can play around with this change and make the appropriate updates to the documentation.

File: tests/test_rests.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        response = self.test_app.get(
+            '/api/rests',
+            follow_redirects=True
+        )
+
+        # Assert response is 200 OK.
+        self.assertEquals(response.status, "200 OK")
+        payload = json.loads(response.data)
+        self.assertEquals(payload['count'], 2)
+        self.assertEquals(payload['rests'][0]['mood_rating_1'], 2)
```
Comment (sienna-kopf) [inline tests/test_rests.py:37]: Nice testing!

**Review by sienna-kopf — APPROVED** [review]
GREAT work on this @Chulstro

**Pull request merged**